src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralService.java

```diff
@@ -68,12 +68,14 @@ private OffenderDeletionReferral createReferral(final OffenderPendingDeletion ev
                 .birthDate(event.getBirthDate())
                 .build();
 
-        event.getOffenders().forEach(offender ->
-                offender.getOffenderBookings().forEach(booking ->
-                        referral.addReferredOffenderBooking(ReferredOffenderBooking.builder()
-                                .offenderId(offender.getOffenderId())
-                                .offenderBookId(booking.getOffenderBookId())
-                                .build())));
+        event.getOffenders().forEach(offender -> {
+
+            final var offenderIds = ReferredOffenderIds.builder().offenderId(offender.getOffenderId());
+
+            offender.getOffenderBookings().forEach(booking -> offenderIds.offenderBookId(booking.getOffenderBookId()));
```

Comment:
not understanding this code.  We're looping over each offender relating to an event.  For each one we're constructing a singular referred offender (variable is `offenderIds` though).  For each offender booking we're then setting the offender book id of the referred offender.  Won't this overwrite each time - why aren't you just calling `last` on the offender bookings and set the referred offending book id to that one - why the loop over the offender bookings?

Reply:
Totally agree, this wouldn't have worked, I've corrected the logic now.